loopingtaskproducer/src/main/java/org/duracloud/mill/ltp/StateManager.java

```diff
@@ -57,8 +61,10 @@ public StateManager(String path, Class<T> klazz) {
     private void flush() {
         ObjectMapper mapper = new ObjectMapper();
         try {
-            mapper.writeValue(this.stateFile, this.state);
+            mapper.writeValue(this.tempStateFile, this.state);
             log.debug("saved {} to {}", this.state, this.stateFile.getAbsolutePath());
+            this.stateFile.delete();
+            FileUtils.moveFile(this.tempStateFile,  this.stateFile);
```

Comment:
Can a flush occur more than once for a StateManager? If so, does anything additional need to be done to prepare the files for the next write after the move?

Reply:
A unit test to make sure this works as expected may be the best approach.